modules/apm/src/main/java/org/elasticsearch/tracing/apm/APMTracer.java

```diff
@@ -452,4 +462,53 @@ private static Automaton patternsToAutomaton(List<String> patterns) {
         }
         return Operations.union(automata);
     }
+
+    public <T> DoubleGauge registerDoubleGauge(MetricName name, String description, T unit) {
+        return doubleGauges.compute(name, (k, v) -> {
+                if (v != null) {
+                    throw new IllegalStateException("double gauge [" + name.getRawName() + "] already registered");
+                }
+                return OtelDoubleGauge.build(services.meter, name, description, unit);
+            }
+        );
+    }
+
+    public void recordDoubleGauge(MetricName name, double value) {
```

Comment:
I suspect that when a gauge/or any instrument is registered it will be used in the same class (contrary to traces/spans where they might start/end in different areas of code)
so we probably don't need those record* methods here. We have those in the DoubleGauge/DoubleHistogram classes.